core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAgentResourceConfigStep.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.deployment.steps;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.JsonReader;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourcePatternsBuildItem;
+import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
+
+public class NativeImageAgentResourceConfigStep {
+    private static final Logger log = Logger.getLogger(NativeImageAgentResourceConfigStep.class);
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    void transformResourceConfig(BuildProducer<NativeImageResourcePatternsBuildItem> resourcePatternProducer,
+            BuildSystemTargetBuildItem buildSystemTargetBuildItem) {
+        final Path agentResourceConfigPath = buildSystemTargetBuildItem.getOutputDirectory()
+                .resolve(Path.of("generated-native-config", "resource-config.json"));
+        if (agentResourceConfigPath.toFile().exists()) {
+            final Pattern ignoreResources = Pattern.compile(
+                    "^((?!(microprofile|quarkus|smallrye|jboss|junit|slf4j|application.properties|surefire|logging.properties|Test.class|java/lang|java/util)).)*$");
```

Comment:
what makes something qualified /necessary to have in this list?

Reply:
Good question.

I created the list based on the output of the native image agent config for the getting started reactive sample JVM mode unit tests, see [here](https://gist.github.com/galderz/2008721d5b04f31bcc40cc67eb0b71e9).

Something more dynamic would likely be needed here. I guess there could be some general starting point for this list, and then based on the extensions in use, expand it?